core/src/main/java/io/questdb/cairo/wal/seq/TableWriterPressureControlImpl.java

```diff
@@ -0,0 +1,154 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.wal.seq;
+
+import io.questdb.cairo.wal.TableWriterPressureControl;
+import io.questdb.std.Rnd;
+import io.questdb.std.datetime.millitime.MillisecondClockImpl;
+
+public class TableWriterPressureControlImpl implements TableWriterPressureControl {
```

Comment:
it would be good to have some background for this class, may be comment on top of it detailing the problem this class solves

Reply:
I added comments in `TableWriterPressureControl` interface and referenced it here.